### PR TITLE
Fixes for opening a connection

### PR DESCRIPTION
**Pull request by jontore**
Main is a singelton, so auth token is shared between imports of the library.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1,66 +1,68 @@
...
+        endpoint: 'mqtt.relayr.io'
+    },
+    ajax: {
+        uri: 'api.relayr.io',
+        dataUri: 'data-api.relayr.io'
+    }
+};
+
+let currentUser;
+let project;
```
Comment (SmbatYeranyan) [inline src/main.js]: Maybe we can make this nicer?

File: src/main.js [hunk 2]
```diff
@@ -1,66 +1,68 @@
...
+    authorize: function(optionalToken) {
         return new Promise((resolve, reject) => {
 
             const oauth2 = new Oauth2({
-                uri: this.config.ajax.uri,
-                appId: this.project.id,
-                redirectURI: this.project.redirectURI,
-                persist: this.config.persistToken
+                uri: config.ajax.uri,
+                appId: project.id,
```
Comment (SmbatYeranyan) [inline src/main.js]: Perhaps we can iterate to a Projects class that holds vital project info redirect_uri and even api calls to list projects

**Comment by SmbatYeranyan** [comment]
Good work on the changing main into something easier to use. Lets think about how we can share Projects nicer. Overall good to go 👍

File: src/main.js (hunk 1 above)
Comment (jontore) [inline src/main.js]: Yes, I think this should go together with gettingc the project config out.

**Pull request merged**